### PR TITLE
[requests] loosen HTTPError constructor

### PR DESCRIPTION
**Pull request by hsheth2**
Follow up on https://github.com/python/typeshed/pull/10766, should help with https://github.com/python/typeshed/issues/10764

### CODE REVIEW

File: stubs/requests/requests/exceptions.pyi [hunk 1]
```diff
@@ -16,7 +16,7 @@ class JSONDecodeError(InvalidJSONError): ...
 class HTTPError(RequestException):
     request: Request | PreparedRequest
```
Comment (srittau) [inline stubs/requests/requests/exceptions.pyi:17]: We also need to add `None` to `request`.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, we need one more change.

File: stubs/requests/requests/exceptions.pyi (hunk 1 above)
Comment (hsheth2) [inline stubs/requests/requests/exceptions.pyi:17]:
```suggestion
    request: Request | PreparedRequest | None
```

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**